src/helpers/createPdfAndSave.js

```diff
@@ -0,0 +1,154 @@
+
+  return newFileMetadata
+}
+
+/**
+ * @param {SaveFileParam} param
+ * @returns {Promise<string>} Filename
+ */
+const saveFile = async ({
+  client,
+  fileCollection,
+  pdfBytes,
+  contacts,
+  opts
+}) => {
+  const { name, metadata, dirId } = opts
+  const { data: fileCreated } = await uploadFileWithConflictStrategy(
+    client,
+    pdfBytes,
+    {
+      name,
+      contentType: 'application/pdf',
+      metadata,
+      dirId,
+      conflictStrategy: 'rename'
+    }
+  )
+
+  // The user is referenced as a contact on the new Paper
+  const references = contacts.map(contact => ({
+    _id: contact._id,
+    _type: CONTACTS_DOCTYPE
+  }))
+  await fileCollection.addReferencedBy(fileCreated, references)
+
+  return fileCreated._id
+}
+
+/**
+ * Convert image & pdf file to pdf & save it
+ *
+ * @param {createAndSavePdfParam} param
+ * @returns {Promise<{ fileId: string, themeLabel: string }[]>} Return array of object with file id & theme label to find its location
+ */
+export const createPdfAndSave = async ({
+  formData,
+  qualification,
+  currentDefinition,
+  appFolderID,
+  client,
+  i18n
+}) => {
+  const { t, f, scannerT } = i18n
+  const { data, metadata, contacts } = { ...formData }
+  const fileCollection = client.collection(FILES_DOCTYPE)
+  const isMultiPage = data.some(({ fileMetadata }) => fileMetadata.multipage)
+  const { featureDate, label } = currentDefinition
+  const date = metadata[featureDate] && f(metadata[featureDate], 'YYYY.MM.DD')
+  let fileCreated = []
+
+  // Created first page of pdf
+  let pdfDoc = await PDFDocument.create()
+
+  for (let idx = 0; idx < data.length; idx++) {
+    const { file, fileMetadata } = data[idx]
+    const pdfBytes = await addFileToPdf(pdfDoc, file)
+
+    const paperName = buildFilename({
+      qualificationName: scannerT(`items.${label}`),
+      pageName: fileMetadata.page
+        ? t(`PapersList.label.${fileMetadata.page}`)
+        : null,
+      username: getFullname(contacts[0]),
+      date
+    })
+
+    // Created metadata for pdf file
+    const newMetadata = {
+      qualification: {
+        ...qualification
+      },
+      ...sanitizeFileMetadata(fileMetadata, isMultiPage),
+      ...metadata,
+      datetime: metadata[featureDate]
+        ? metadata[featureDate]
+        : pdfDoc.getCreationDate(),
+      datetimeLabel: metadata[featureDate] ? featureDate : 'datetime'
+    }
+
+    // If isn't multipage or the last of multipage, save file
+    if (!isMultiPage || (isMultiPage && idx === data.length - 1)) {
+      const fileIdCreated = await saveFile({
+        client,
+        fileCollection,
+        pdfBytes,
+        contacts,
+        opts: {
+          name: paperName,
+          metadata: newMetadata,
+          dirId: appFolderID
+        }
+      })
+      fileCreated.push({ fileId: fileIdCreated, themeLabel: label })
+    }
+
+    // If isn't multipage & not the last page, create new page of PDF
```

Comment:
`PDFDocument.create()` is not about `page` creation but about `document` creation. So I'm not sure to understand the comment.

Reply:
This seems to be related to the ambiguity between pdf page and form page. Isn't it?

Reply:
Indeed, I renamed it